cmd/oras/pull.go

```diff
@@ -101,15 +101,22 @@ func runPull(opts pullOptions) error {
 
 	// Copy Options
 	copyOptions := oras.DefaultCopyOptions
-	configPath, configMediaType := parseFileReference(opts.ManifestConfigRef, oras.MediaTypeUnknownConfig)
+	configPath, configMediaType := parseFileReference(opts.ManifestConfigRef, "")
 	copyOptions.FindSuccessors = func(ctx context.Context, fetcher content.Fetcher, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
 		successors, err := content.Successors(ctx, fetcher, desc)
 		if err != nil {
 			return nil, err
 		}
 		var ret []ocispec.Descriptor
-		for _, s := range successors {
-			if s.MediaType == configMediaType {
+		for i, s := range successors {
+			// Save the config when:
+			// 1) MediaType matches, or
+			// 2) MediaType not specified and current node is config.
+			// Note: For a manifest, the 0th indexed element is always a
+			// manifest config.
+			if s.MediaType == configMediaType || (configMediaType == "" && i == 0 &&
+				(desc.MediaType == "application/vnd.docker.distribution.manifest.v2+json" ||
+					desc.MediaType == ocispec.MediaTypeImageManifest)) {
```

Comment:
Let's make `application/vnd.docker.distribution.manifest.v2+json` a constant and make this Boolean expression as a function.

Reply:
Made requested change